lucille-core/src/main/java/com/kmwllc/lucille/stage/AddRandomNestedField.java

```diff
@@ -130,32 +132,48 @@ public AddRandomNestedField(Config config) throws StageException {
   // Start generator stages
   @Override
   public void start() throws StageException {
-    if (generatorsConfig == null) {
-      return;
-    }
+    if (generatorsConfig != null) {
+      // Iterate over each generator param
+      for (String key : generatorsConfig.root().keySet()) {
+        Config sub = generatorsConfig.getConfig(key);
+        if (!sub.hasPath("class")) {
+          throw new StageException("generators." + key + " must include a 'class' property");
+        }
 
-    // Iterate over each generator param
-    for (String key : generatorsConfig.root().keySet()) {
-      Config sub = generatorsConfig.getConfig(key);
-      if (!sub.hasPath("class")) {
-        throw new StageException("generators." + key + " must include a 'class' property");
-      }
+        Config injected = ConfigFactory.parseMap(Map.of(
+            "name", "arnf_gen_" + key,
+            "field_name", GEN_OUT_FIELD
+        )).withFallback(sub);
 
-      Config injected = ConfigFactory.parseMap(Map.of(
-          "name", "arnf_gen_" + key,
-          "field_name", GEN_OUT_FIELD
-      )).withFallback(sub);
+        // Create generator
+        Stage gen;
 
-      // Create generator
-      Stage gen;
+        try {
+          gen = Stage.fromConfig(injected);
+        } catch (Exception e) {
+          throw new StageException("Failed to instantiate generator '" + key + "'", e);
+        }
+        gen.start();
+        generators.put(key, gen);
+      }
+    }
 
-      try {
-        gen = Stage.fromConfig(injected);
-      } catch (Exception e) {
-        throw new StageException("Failed to instantiate generator '" + key + "'", e);
+    // create set of destination field parts so we don't have to split them on every doc
+    final int n = pickNumObjects();
+    int index = 0;
```

Comment:
why do we have to maintain this `index` separately from the `i` in the for loop?